resources/testharness.js

```diff
@@ -1669,20 +2042,13 @@
         }
     }
 
+    // TODO: Figure out how to document the overloads better.
```

Comment:
Glad I came across this TODO. I also wondered when reading some of the `thisOrThat` parameter names, if there isn't a way to describe the valid function signatures, more like what MDN does now: https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#syntax

But I guess that's not very JS-y and it's not too surprising if it's not supported.

Reply:
So JSDoc kind of supports overloads c.f. https://github.com/jsdoc/jsdoc/issues/1017 but I couldn't get it working with sphinx-js (c.f. https://github.com/mozilla/sphinx-js/issues/152 although that's about typescript). This was definitely about the point I started to consider the merits of writing a replacement for sphinx-js ;)